xarray/tests/test_coding.py

```diff
@@ -20,20 +22,36 @@ def test_CFMaskCoder_decode():
     assert_identical(expected, encoded)
 
 
-def test_CFMaskCoder_encode_missing_fill_values_conflict():
-    original = xr.Variable(
-        ("x",),
-        [0.0, -1.0, 1.0],
-        encoding={"_FillValue": np.float32(1e20), "missing_value": np.float64(1e20)},
-    )
-    coder = variables.CFMaskCoder()
-    encoded = coder.encode(original)
+encoding_with_dtype = {
+    "dtype": np.dtype("float64"),
```

Comment:
What happens when  `dtype=np.dtype("int64")` with float missing & fill values?

Reply:
As in #3502, an attempt is made to cast the missing/fill values to `int64`.  Should we do something different?

Reply:
LGTM. sorry for the noise.